### ISSUE TITLE
[Bug]: Dialog in v9 doesn't work with React.ScrtictMode

### ISSUE THREAD

**Issue by XFox111** [microsoft/fluentui#24423]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 5.4 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
    Memory: 223.71 MB / 7.77 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/fluentui9-dialog-repro-d839cb

## Actual Behavior
When app is wrapped into `<React.StrictMode>` tag the dialog is not working. Took some time to figure this out since yarn's typescript template places it automatically in index.tsx

## Expected Behavior
Dialog should open when `<React.StrictMode>` is presented or documentation should contain warning about this issue

### Requested priority

Low

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by bsunderhus** [comment]
Just to clarify, this is not about `React.StrictMode`, it's about React@18.

As you can see in the [following example](https://codesandbox.io/s/fluentui9-dialog-repro-forked-7kjep2?file=/src/index.js), `Dialog` works fine with `StrictMode`at React@17. We currently do not support React@18 for v9, but nonetheless this is a good bug worth investigating.

**Comment by bsunderhus** [comment]
We're working on it @XFox111! We hope to have React@18 supported in no time! There are a couple of bumps in the way of supporting concurrent mode, but we're almost there!

**Comment by XFox111** [comment]
Btw, it seems that the issue is true for any component in v9 that utilizes overlay. I've just tested it and it's there for `Popover`, `Combobox` and `Tooltip` as well. Thought it might be useful for you to know

**Comment by AndrewCraswell** [comment]
@bsunderhus I second the idea to add a mention about React 18 support to the v9 docs, probably under the Migration section. Now that v8 supports React 18, I ran into issues trying to upgrade. I hadn't realized v9 doesn't support it, and I couldn't find any mention. I did manage to discover that using any v9 component is what was causing my app to crash.

![image](https://user-images.githubusercontent.com/1908292/206856854-73cec4fd-0816-4ef2-8d96-9f1fa353813e.png)

I spent way longer than I would like to admit troubleshooting this, and identifying v9 as the culprit. So I vote calling it out explicitly in the migration guide, since it may impact someone's decision to migrate from v8 to v9, if it would require the rollback of React 18 to 17.

**Comment by layershifter** [comment]
> Btw, it seems that the issue is true for any component in v9 that utilizes overlay. I've just tested it and it's there for `Popover`, `Combobox` and `Tooltip` as well. Thought it might be useful for you to know

This was a problem with `Portal`, fixed by #25956.

https://codesandbox.io/s/fluentui9-dialog-repro-forked-g3dblq?file=/src/App.js
